libevmasm/GasMeter.cpp

```diff
@@ -182,7 +180,11 @@ GasMeter::GasConsumption GasMeter::estimateMax(AssemblyItem const& _item, bool _
 		case Instruction::EXP:
 			gas = GasCosts::expGas;
 			if (u256 const* value = classes.knownConstant(m_state->relativeStackElement(-1)))
-				gas += GasCosts::expByteGas(m_evmVersion) * (32 - (h256(*value).firstBitSet() / 8));
+			{
+				// Note: msb() counts from 0.
+				auto const significant_byte_count = (boost::multiprecision::msb(*value) + 1 + 7) / 8;
```

Comment:
What will ``boost::multiprecision::msb`` for ``u256(0)``? IIRC it will throw, right? We don't want that, do we?

Reply:
Will it throw? I just couldn't find any documentation on it.

Reply:
boost's documentation is really horrible :-)...

Reply:
https://www.boost.org/doc/libs/1_72_0/libs/multiprecision/doc/html/boost_multiprecision/tut/gen_int.html
says something about " Throws a std::domain_error if x <= 0. "

Reply:
But I haven't checked...

Reply:
If it does, we don't have a clear test for it.

Update: added a test case to trigger and indeed it throws.